### DISCUSSION TITLE
os.iorun os.exec 等 api 不支持管道

### DISCUSSION THREAD

**Discussion by CN-Zxcv** [xmake-io/xmake#6002] [Q&A]
### Xmake 版本

2.9.6

### 操作系统版本和架构

rockylinux9

### 描述问题

local out, err = os.iorun("gcc --version | head -1")
期望是只得到第一行，但是返回了整个输出

### 期待的结果

管道或者其他替代方案

### 工程配置

target("test")

### 附加信息和错误日志

target('test')                                      
    before_build(function (target)                          
        local out, err = os.iorun("gcc --version | head -1")
        print(out, err)                                     
    end)                                                    
    on_build(function()                                     
    end)

[a comment by an automated account is not shown]

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake/blob/4f408ebd0c944fecb9777c5b07ec57af1f5798b2/xmake/modules/private/service/remote_build/server_session.lua#L264
